### DISCUSSION TITLE
Redirect to new URL

### DISCUSSION THREAD

**Discussion by davisums** [unpoly/unpoly#522] [Q&A] [answered]
### Bug description

This isn't really a bug, but more something that might be missing from the API. I haven't found any way to redirect to a new URL, either from the client or the server. Specifically my scenario is:

1.  User is on `page/[id]/processing
2. There's an up-poll element on the page
3. Something happens on the server side and I want the user to be redirected to a new URL

I was thinking of two ways to do this:

- Either some kind of new up-follow
- A custom header that redirects to a new URL. I thought X-Up-Location would do this, but it doesn't
- Some way to do this with up-history?

From my tests, none of these is currently possible. Any ideas how to do this or any API plans to support something like this?

**Comment by adam12** [comment, marked as answer]
I haven't tested this, but I might be inclined to emit a [custom Unpoly event](https://unpoly.com/X-Up-Events) via Header and have a listener that will do the navigation for you.

```
up.on('poll:redirect-request', (ev) {
  up.navigate({url: ev.location})
})

# In a header
X-Up-Events: [{"type": "poll:redirect-request", "location: "/new-url"}]

**Comment by davisums** [comment]
Great idea, that worked exactly as expected. I still think some kind of built-in support for redirects would be nice to have though.

Thanks for the help!
